VelocidiSDK/VelocidiSDK/VSDKUtil.m

```diff
@@ -0,0 +1,63 @@
+    userAgent = [NSString stringWithFormat:@"%@/%@ %@/%@ (%@; iOS %@; Scale/%0.2f)",
+                 [[NSBundle mainBundle] infoDictionary][(__bridge NSString *)kCFBundleExecutableKey] ?: [[NSBundle mainBundle] infoDictionary][(__bridge NSString *)kCFBundleIdentifierKey],
+                 [[NSBundle mainBundle] infoDictionary][@"CFBundleShortVersionString"] ?: [[NSBundle mainBundle] infoDictionary][(__bridge NSString *)kCFBundleVersionKey],
+                 @"VelocidiSDK",
+                 @(VelocidiSDKVersionNumber),
+                 [[UIDevice currentDevice] model],
+                 [[UIDevice currentDevice] systemVersion],
+                 [[UIScreen mainScreen] scale]];
+#elif TARGET_OS_WATCH
+    // User-Agent Header; see http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.43
+    userAgent = [NSString stringWithFormat:@"%@/%@ %@/%@ (%@; watchOS %@; Scale/%0.2f)",
+                 [[NSBundle mainBundle] infoDictionary][(__bridge NSString *)kCFBundleExecutableKey] ?: [[NSBundle mainBundle] infoDictionary][(__bridge NSString *)kCFBundleIdentifierKey],
+                 [[NSBundle mainBundle] infoDictionary][@"CFBundleShortVersionString"] ?: [[NSBundle mainBundle] infoDictionary][(__bridge NSString *)kCFBundleVersionKey],
+                 @"VelocidiSDK",
+                 @(VelocidiSDKVersionNumber),
+                 [[WKInterfaceDevice currentDevice] model],
+                 [[WKInterfaceDevice currentDevice] systemVersion],
+                 [[WKInterfaceDevice currentDevice] screenScale]];
+#elif defined(__MAC_OS_X_VERSION_MIN_REQUIRED)
+    userAgent = [NSString stringWithFormat:@"%@/%@ %@/%@ (Mac OS X %@)",
+                 [[NSBundle mainBundle] infoDictionary][(__bridge NSString *)kCFBundleExecutableKey] ?: [[NSBundle mainBundle] infoDictionary][(__bridge NSString *)kCFBundleIdentifierKey],
+                 [[NSBundle mainBundle] infoDictionary][@"CFBundleShortVersionString"] ?: [[NSBundle mainBundle] infoDictionary][(__bridge NSString *)kCFBundleVersionKey],
+                 @"VelocidiSDK",
+                 @(VelocidiSDKVersionNumber),
+                 [[NSProcessInfo processInfo] operatingSystemVersionString]];
+#endif
+    if (userAgent) {
+        if (![userAgent canBeConvertedToEncoding:NSASCIIStringEncoding]) {
+            NSMutableString *mutableUserAgent = [userAgent mutableCopy];
+            if (CFStringTransform((__bridge CFMutableStringRef)(mutableUserAgent), NULL, (__bridge CFStringRef)@"Any-Latin; Latin-ASCII; [:^ASCII:] Remove", false)) {
+                userAgent = mutableUserAgent;
+            }
+        }
+    }
+    return userAgent;
+    
+}
+
+- (NSUUID *)getAdvertisingIdentifier {
+    return [[ASIdentifierManager sharedManager] advertisingIdentifier];
+}
+
+- (bool)isTrackAllowed:(NSUUID *)advertisingIdentifier {
+    if ([[advertisingIdentifier UUIDString] isEqualToString:@"00000000-0000-0000-0000-000000000000"]){
```

Comment:
Why not use [`isAdvertisingTrackingEnabled `](https://developer.apple.com/documentation/adsupport/asidentifiermanager/1614148-isadvertisingtrackingenabled)?

Reply:
Tracking is more something that the user allows or disallows, not soo much as something that is enabled or disabled 😅.

Reply:
Nvm, I was not aware of https://developer.apple.com/documentation/adsupport/asidentifiermanager/1614148-isadvertisingtrackingenabled. I thought that you were suggesting renaming that function. Using a native function is indeed much better, I totally missed that one.

Reply:
Changed it in 81466fd. I still left the function in VSDKUtil, so that it is easier to mock and test it. Let me know what you think about that. I know that it is a bit silly to have all that just to test an `if` but I still think it would be good to have a test that verifies that we are not tracking when tracking is not allowed.

Reply:
I think we can improve this a little bit.

Why don't we have a private property named `sharedManager` of type `ASIdentifierManager`? When using it, if it is `nil`, then we set it to `[ASIdentifierManager sharedManager]`. This allows us to create a mock class which implements `ASIdentifierManager` and we set it to `sharedManager` (the new property we created).

I haven't tested this, but it should work. I'm only worried if implementing/extending `ASIdentifierManager` is possible.

Reply:
> I think we can improve this a little bit.
> 
> Why don't we have a private property named `sharedManager` of type `ASIdentifierManager`? When using it, if it is `nil`, then we set it to `[ASIdentifierManager sharedManager]`. This allows us to create a mock class which implements `ASIdentifierManager` and we set it to `sharedManager` (the new property we created).
> 
> I haven't tested this, but it should work. I'm only worried if implementing/extending `ASIdentifierManager` is possible.

@duartepinto What do you think about this?

Reply:
Sorry, overlooked this. On it

Reply:
Implemented this in b8e520a